### PR TITLE
chore: Angular 16 support

### LINKED ISSUE #588: Add support for v16 in @uirouter/angular-hybrid
As of right now, the latest supported Angular version is 15. We're in the process of upgrading from 15 to 16, and this is blocking us right now.

### PR DESCRIPTION
**Pull request by boxmein**
This PR implements Angular 16 support. 

Once the sample app (https://github.com/ui-router/sample-app-angular-hybrid/pull/914) is updated, this PR is also mergeable

Fixes #588

### CODE REVIEW

**Comment by boxmein** [comment]
Taking care of https://github.com/ui-router/sample-app-angular-hybrid/issues/913

**Comment by wawyed** [comment]
hmm looks like the example is failing.

File: example/package.json
Comment (wawyed) [inline example/package.json:1]: Might want to update

    "rxjs": "^7.4.0",
    "typescript": "4.8.4",
    "zone.js": "0.11.4"
    
    
as well

**Comment by wawyed** [comment]
I figure it out, you need to update:

    "webpack": "^5.77.0",
    "webpack-cli": "^5.0.1",
    "webpack-dev-server": "^4.13.2"
    
    in the examples package.json

**Comment by boxmein** [comment]
Thanks so much. If it's OK I will pick it up after my vacation, at the 2nd week of August.

**Comment by boxmein** [comment]
Testing locally, with your recommended changes, the `test:downstream` NPM script started passing.

**Comment by wawyed** [comment]
Hi, can you change the downstream test to uirouter I just merged it.

**Comment by boxmein** [comment]
Thanks @wawyed, I'll take care of it.

**Pull request merged**